Multisig/Features/Security/KeychainStorage.swift

```diff
@@ -0,0 +1,240 @@
+        }
+    }
+
+    private func convertStatusToError(status: OSStatus) -> Error {
+        let message = SecCopyErrorMessageString(status, nil) as? String ?? String(status)
+        return NSError(domain: NSOSStatusErrorDomain, code: Int(status), userInfo: [NSLocalizedDescriptionKey: message])
+    }
+
+    private func findPasswordData() throws -> Data? {
+        let query = SQuery.generic(id: KeychainStorage.derivedPasswordTag).queryData()
+
+        var item: CFTypeRef?
+        let status = SecItemCopyMatching(query, &item)
+
+        switch status {
+        case errSecSuccess:
+            if let item = item {
+                return item as? Data
+            } else {
+                return nil
+            }
+
+        case errSecItemNotFound:
+            return nil
+
+        case let status:
+            throw convertStatusToError(status: status)
+        }
+    }
+
+    // used to create KEK
+    func createSecureEnclaveKey(
+            useBiometry: Bool,
+            canChangeBiometry: Bool,
+            applicationPassword: String
+    ) throws -> SecKey {
+
+        //create flags from booleans
+        var flags: SecAccessControlCreateFlags = []
+        if canChangeBiometry && useBiometry {
+            flags = [.biometryAny, .or, .devicePasscode, .applicationPassword]
+        }
+        if !canChangeBiometry && useBiometry {
+            flags = [.biometryCurrentSet, .or, .devicePasscode, .applicationPassword]
+        }
+        if !useBiometry {
+            flags = [.applicationPassword]
+        }
+        return try createSEKey(flags: flags, applicationPassword: applicationPassword)
+    }
+
+    func storeData(encryptedData: Data, account: String) throws {
+        // delete existing account data
+        try deleteData(account)
+
+        // Create query
+        let addEncryptedDataQuery = SQuery.generic(id: account, encryptedData: encryptedData).queryData()
+
+        // safe to Keychain (as type password?) using SecItemAdd()
+        let status = SecItemAdd(addEncryptedDataQuery, nil)
+
+        switch status {
+        case errSecSuccess:
+            return
+        case let status:
+            throw convertStatusToError(status: status)
+        }
+    }
+
+    func retrieveEncryptedData(account: String) throws -> Data? {
+        let query = SQuery.generic(id: account).queryData()
+
+        var item: CFTypeRef?
+        let status = SecItemCopyMatching(query, &item)
+
+        switch status {
+        case errSecSuccess:
+            if let item = item {
+                return item as? Data
+            } else {
+                return nil
+            }
+
+        case errSecItemNotFound:
+            return nil
+
+        case let status:
+            throw convertStatusToError(status: status)
+        }
+    }
+
+    func deleteData(_ account: String) throws {
+        let query = SQuery.generic(id: account).queryData()
+
+        let status = SecItemDelete(query)
+        switch status {
```

Comment:
I think it could be replaced with this:
```
        if status != errSecSuccess && status != errSecItemNotFound {
            throw convertStatusToError(status: status)
        }
```
Which is more readabla, nor?

Reply:
Or  with:
```
guard status == errSecSuccess || status == errSecItemNotFound else {
            throw convertStatusToError(status: status)
   }
```

Which is even more readable because it is not using a negation.